From ff0f8382972b18c3c5e873d869ceb4adb821ddf7 Mon Sep 17 00:00:00 2001
From: Enrico Scholz <enrico.scholz@informatik.tu-chemnitz.de>
Date: Sat, 2 Mar 2013 17:31:53 +0100
Subject: [PATCH] remove timer fd explicitly from epoll

Forked processes can keep the old fd alive triggering epoll over and
over again else.


Signed-off-by: Enrico Scholz <enrico.scholz@informatik.tu-chemnitz.de>

---
 src/core/manager.c | 2 ++
 1 Datei geändert, 2 Zeilen hinzugefügt(+)

diff --git a/src/core/manager.c b/src/core/manager.c
index 1ddd8ba..57c622b 100644
--- a/src/core/manager.c
+++ b/src/core/manager.c
@@ -1479,6 +1479,8 @@ static int process_event(Manager *m, struct epoll_event *ev) {
                            NULL);
 
                 /* Restart the watch */
+                epoll_ctl(m->epoll_fd, EPOLL_CTL_DEL, m->time_change_watch.fd,
+                          NULL);
                 close_nointr_nofail(m->time_change_watch.fd);
                 watch_init(&m->time_change_watch);
                 manager_setup_time_change(m);
-- 
1.7.11.7

